iiasa/ixmp
# Suppress Java noise on Windows/AppVeyor

Closes #229.

See jpype-project/jpype#561. To summarize:
- pytest has a core plugin that enables Python's `faulthandler` module by default.
- Something in Java causes signals to be emitted, causing these tracebacks to be printed by `faulthandler`. However, per [this comment](https://github.com/jpype-project/jpype/issues/561#issuecomment-573476302), these are intentional, and not fatal in Java—so the messages are spurious and not related to any error/failure.

This PR
- Addresses the issue by adding `-p no:faulthandler` to the pytest options.
- Also simplifies ixmp.backend.jdbc.start_jvm, allowing JPype to construct the classpath argument.

## How to review

- View the AppVeyor build logs for the latest commit on this branch and confirm that the messages described in #229 do not appear.
- Run on a Windows machine and confirm again that the messages do not appear.

## PR checklist

- [x] ~Tests added.~ N/A
- [x] ~Documentation added.~ N/A, CI/internal changes only.
- [x] ~Release notes updated.~ N/A, CI/internal changes only.
